master/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserService.java

```diff
@@ -472,6 +474,15 @@ private User fromEntity(UserDescriptor userDescriptor) throws BadRequestExceptio
         return user;
     }
 
+    private void checkUserName(String name) throws BadRequestException {
+        if (isNullOrEmpty(name)) {
+            throw new BadRequestException("User name required");
+        }
+        if (!VALID_USER_NAME.matcher(name.toLowerCase()).matches()) {
```

Comment:
It's problem of variable name because name will be converted to lower case. But maybe It would be better to convert username to lower case before using instead of checking with uppercase letters and converting after that